### PR TITLE
Remove transition code from DelayedJob

### PR DESCRIPTION
**Pull request by mkllnk**
And a bit more cleaning.

We disabled DelayedJob a long time ago. We can consider all servers updated or an older version needs to be used.

### CODE REVIEW

**Review by jibees — APPROVED** [review]
Great news this kind of cleaning 🧹 

Thanks for that @mkllnk !

**Comment by jibees** [comment]
Is there any way to test this one? I guess not (and we should merge directly) but i'm wondering. 

Maybe it needs a second review as well.

**Comment by mkllnk** [comment]
We could provision a staging server and see that it still works. I think I did that already. I'm requesting a second review but then I think that we can merge it.

File: roles/deploy/handlers/main.yml
Comment (dacook) [inline roles/deploy/handlers/main.yml:48]: TIL about having a list of `when` conditions.

**Review by dacook — APPROVED** [review]
👍

**Comment by dacook** [comment]
**Test deployment**: ✅

```
...
RUNNING HANDLER [deploy : restart sidekiq] *****************************************************************************
changed: [staging.openfoodnetwork.org.au]
Tuesday 07 February 2023  09:56:46 +1100 (0:00:13.021)       0:04:35.971 ******
```

**Pull request merged**
